Source/Scene/createTileMapServiceImageryProvider.js

```diff
@@ -175,30 +167,30 @@ define([
             var rectangle = Rectangle.clone(options.rectangle);
 
             if (!defined(rectangle)) {
-                var swXY;
-                var neXY;
                 var sw;
                 var ne;
+                var swXY = new Cartesian2(parseFloat(bbox.getAttribute('minx')), parseFloat(bbox.getAttribute('miny')));
+                var neXY = new Cartesian2(parseFloat(bbox.getAttribute('maxx')), parseFloat(bbox.getAttribute('maxy')));
 
-                if (flipXY) {
-                    swXY = new Cartesian2(parseFloat(bbox.getAttribute('miny')), parseFloat(bbox.getAttribute('minx')));
-                    neXY = new Cartesian2(parseFloat(bbox.getAttribute('maxy')), parseFloat(bbox.getAttribute('maxx')));
-
-                    // In old tilers with X/Y flipped, coordinate are always geodetic degrees.
+                // Determine based on the profile attribute if this tileset was generated by gdal2tiles.py, which
+                // uses 'mercator' and 'geodetic' profiles, or by a tool compliant with the TMS standard, which is
+                // 'global-mercator' and 'global-geodetic' profiles.
+                //
+                // In the gdal2Tiles case, X and Y are always in geodetic degrees. In older versions of gdal (and only,
+                // gdal) they were also flipped, which is why we check for an option to flip the values here as well.
+                // Unfortunately there is no way to autodetect whether flipping is needed or not.
+                var isGdal2tiles = tilingSchemeName === 'geodetic' || tilingSchemeName === 'mercator';
+                var flipXY = defaultValue(options.flipXY, false);
+                if (isGdal2tiles && flipXY) {
```

Comment:
If you're going to provide an `options.flipXY`, you should make it flip all the time, not just for a tileset that was apparently produced by gdal2tiles.py.  Or, at least, the doc needs to be very clear that the option is ignored if we don't think it came from gdal2tiles.py.